src/server/editorServices.ts

```diff
@@ -1251,7 +1284,10 @@ namespace ts.server {
                     return error;
                 }
                 else {
-                    const oldFileNames = project.compilerService.host.roots.map(info => info.fileName);
+                    // if the project is too large, the root files might not have been all loaded if the total
+                    // program size reached the upper limit. In that case project.projectOptions.files should 
+                    // be more precise. However this would only happen for configured project.
+                    const oldFileNames = project.projectOptions ? project.projectOptions.files : project.compilerService.host.roots.map(info => info.fileName);

```

Comment:
Just because `projectOptions` were specified, does this mean the files were listed? Or is this different to the `files` setting in the config file? (I may just be confused).